src/algebra/scalar/authenticated_scalar.rs

```diff
@@ -364,49 +364,47 @@ impl<C: CurveGroup> AuthenticatedScalarResult<C> {
         // --- Exchange the MAC Checks and Commitment Blinders --- //
 
         let peer_mac_checks = fabric.exchange_values(&mac_checks);
-        let peer_blinders = fabric.exchange_values(
-            &my_comms.iter().map(|comm| fabric.allocate_scalar(comm.blinder)).collect_vec(),
-        );
+        let peer_blinder = fabric.exchange_value(fabric.allocate_scalar(my_comm.blinder));
 
         // --- Check the MAC Checks --- //
 
-        let mut mac_check_gate_deps = my_comms.iter().map(|comm| comm.value.id).collect_vec();
+        let mut mac_check_gate_deps = my_comm.values.iter().map(|v| v.id()).collect_vec();
         mac_check_gate_deps.push(peer_mac_checks.id);
-        mac_check_gate_deps.push(peer_blinders.id);
-        mac_check_gate_deps.push(peer_comms.id);
+        mac_check_gate_deps.push(peer_blinder.id);
+        mac_check_gate_deps.push(peer_comm.id);
 
         let commitment_checks: Vec<ScalarResult<C>> = fabric.new_batch_gate_op(
             mac_check_gate_deps,
             n, // output_arity
             move |mut args| {
                 let my_comms: Vec<Scalar<C>> = args.drain(..n).map(|comm| comm.into()).collect();
                 let peer_mac_checks: Vec<Scalar<C>> = args.remove(0).into();
-                let peer_blinders: Vec<Scalar<C>> = args.remove(0).into();
-                let peer_comms: Vec<CurvePoint<C>> = args.remove(0).into();
+                let peer_blinder: Scalar<C> = args.remove(0).into();
+                let peer_comm: Scalar<C> = args.remove(0).into();
+
+                // Verify a commitment to the openings
+                let comm = HashCommitment {
+                    values: peer_mac_checks.clone(),
+                    blinder: peer_blinder,
+                    commitment: peer_comm,
+                };
+                if !comm.verify() {
+                    return vec![ResultValue::Scalar(Scalar::zero()); n];
+                }
 
                 // Build a commitment from the gate inputs
                 let mut mac_checks = Vec::with_capacity(n);
-                for (my_mac_share, peer_mac_share, peer_blinder, peer_commitment) in izip!(
-                    my_comms.into_iter(),
-                    peer_mac_checks.into_iter(),
-                    peer_blinders.into_iter(),
-                    peer_comms.into_iter()
-                ) {
-                    let mac_check = Self::verify_mac_check(
-                        my_mac_share,
-                        peer_mac_share,
-                        peer_commitment,
-                        peer_blinder,
-                    );
-                    mac_checks.push(ResultValue::Scalar(Scalar::from(mac_check)));
+                for (my_mac_share, peer_mac_share) in
+                    izip!(my_comms.into_iter(), peer_mac_checks.into_iter(),)
+                {
+                    let sum_to_zero = my_mac_share + peer_mac_share == Scalar::zero();
+                    mac_checks.push(ResultValue::Scalar(sum_to_zero.into()))
```

Comment:
thoughts on re-packaging this into a `verify_mac_check_batch` method (or similar) just so it reads similarly to `open_authenticated_batch` for `CurvePoint`? (just a nit, really)

Reply:
I like it